src/app/components/charts/chart-volume/chart-volume.component.ts

```diff
@@ -0,0 +1,70 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core'
+import * as d3 from 'd3'
+
+import { ChartBaseComponent } from '../chart-base/chart-base.component'
+
+@Component({
+  selector: 'app-chart-volume',
+  templateUrl: '../charts.common.html',
+  styleUrls: ['./chart-volume.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ChartVolumeComponent extends ChartBaseComponent {
+  dates: Date[]
+
+  svg: any
+  context: any
+  chart: any
+  width: number
+  height: number
+  xAxis: any
+  xScale: any
+  yScale: any
+  xScaleBrush: any
+  zoom: any
+  brush: any
+  area: any
+
+  init() {
+    this.context = this.svg
+      .append('g')
+      .attr('class', 'context')
+      .attr('transform', `translate(${this.margin.left}, 0)`)
+
+    super.init()
+  }
+
+  draw() {
+    this.yScale = d3
+      .scaleLinear()
+      .rangeRound([this.height - 25, 0])
+      .domain(d3.extent(this.data, d => d.value as number))
+
+    this.xScale = d3
+      .scaleTime()
+      .rangeRound([0, this.width])
+      .domain(d3.extent(this.data, d => d.date))
+      .nice()
+
+    this.hasYAxis &&
+      this.yAxis.call(d3.axisLeft(this.yScale).tickSize(-this.width))
+
+    this.hasXAxis &&
+      this.xAxis
+        .attr('transform', `translate(0, ${this.height})`)
+        .call(d3.axisBottom(this.xScale))
```

Comment:
The xAxis needs to be customised to look like the designs:
![screen shot 2017-09-21 at 12 05 23](https://user-images.githubusercontent.com/125300/30688860-f663b9c4-9ec7-11e7-854d-7647c8cb1f3e.png)